packages/opentelemetry-core/src/platform/node/sampling-probability.ts

```diff
@@ -20,11 +20,8 @@
  * @returns sampling probability.
  */
 export function getSamplingProbability(): number {
-  const probability = Number(process.env.OTEL_SAMPLING_PROBABILITY || 1);
-  // user can supply an invalid value that when passed to Number returns NaN
-  if (isNaN(probability)) {
-    return 1;
-  }
+  const probability = Number(process.env.OTEL_SAMPLING_PROBABILITY) || 1;
```

Comment:
If users supplies 0, this will make the probability 1

Reply:
I stand corrected... commit reverted